include/soci/soci-platform.h

```diff
@@ -8,6 +8,18 @@
 #ifndef SOCI_PLATFORM_H_INCLUDED
 #define SOCI_PLATFORM_H_INCLUDED
 
+//disable MSVC deprecated warnings
+#ifdef _MSC_VER
+#define _CRT_SECURE_NO_WARNINGS
```

Comment:
Could you please add `#ifndef _CRT_SECURE_NO_WARNINGS` here to avoid redefining it if it's already predefined in the project file (we do define it already on ours)?